### PR TITLE
feat: Add a login page for use with the proxy

### PR DESCRIPTION
**Pull request by tofumatt**
## Summary

Addresses issue #642.

Adds a setup screen that links the user to the proxy, if the proxy is enabled.

## Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.4.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: assets/js/components/setup/setup-proxy.js [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+														mdc-layout-grid__cell--span-12
+													">
+														<h1 className="googlesitekit-setup__title">
+															{ __( 'Time to set up the Site Kit plugin', 'google-site-kit' ) }
+														</h1>
+														<p className="googlesitekit-setup__description">
+															{ __( 'Site Kit Service will guide you through 3 simple setup steps.', 'google-site-kit' ) }
+														</p>
+														<Button
+															href={ proxySetupURL }
```
Comment (aaemnnosttv) [inline assets/js/components/setup/setup-proxy.js]: I think this line is not doing anything since Buttons don't have `href`s 😄 

```suggestion
```

Comment (tofumatt) [reply assets/js/components/setup/setup-proxy.js]: Actually it definitely is 😄. It creates a button-styled link that is marked up as `<a href={url}>` but visually as a button. It's what we want here:

Screenshot 2019-10-18 14 33 41

File: assets/js/components/setup/setup-proxy.js [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+															onClick={ () => {
+																sendAnalyticsTrackingEvent( 'plugin_setup', 'signin_with_google' );
+																document.location = connectUrl;
```
[the remark this answers is not in the document]
Comment (tofumatt) [reply assets/js/components/setup/setup-proxy.js]: Oh, good point. It's `proxySetupURL` but I'll remove the other code to clarify. Copypasta, bad Matt.

**Comment by tofumatt** [comment]
The E2E tests in the main auth branch were failing here when I started, so I thought that was to be expected, but I'll see about regenerating any visual tests and checking on the E2E ones 👍

Regarding the heading, I still think "The Site Kit plugin is active but requires setup" is a confusing header, as mentioned in the issue. I've made the change, but I think a more instructive, actionable header would be clearer and also drive users to complete the setup. They already activated the plugin to get to this page, so I think we're repeating info they already know with the proposed text 😄

**Comment by tofumatt** [comment]
This is ready for another review! 😄

**Comment by felixarntz** [comment]
> Regarding the heading, I still think "The Site Kit plugin is active but requires setup" is a confusing header, as mentioned in the issue. I've made the change, but I think a more instructive, actionable header would be clearer and also drive users to complete the setup. They already activated the plugin to get to this page, so I think we're repeating info they already know with the proposed text

cc @marrrmarrr

**Pull request merged**